### PR TITLE
[zh-cn]: update the translation of WeakMap overview

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#16726

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 1]
```diff
@@ -106,11 +260,11 @@ class ClearableWeakMap {
...
 
-## 相关链接
+## 参见
 
-- A polyfill of `WeakMap` is available in [`core-js`](https://github.com/zloirock/core-js#weakmap)
-- [`WeakMap` in the JavaScript guide](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#WeakMap_object)
-- [Hiding Implementation Details with ECMAScript 6 WeakMaps](http://fitzgeraldnick.com/weblog/53/)
+- [Polyfill of `WeakMap` in `core-js`](https://github.com/zloirock/core-js#weakmap)
+- [Keyed collections](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap_object)
+- [Hiding Implementation Details with ECMAScript 6 WeakMaps](https://fitzgeraldnick.com/2014/01/13/hiding-implementation-details-with-e6-weakmaps.html) by Nick Fitzgerald (2014)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:267]:
```suggestion
- [`core-js` 中 `WeakMap` 的 polyfill](https://github.com/zloirock/core-js#weakmap)
- [带键的集合](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap_对象)
- [使用 ECMAScript 6 WeakMap 隐藏实现细节](https://fitzgeraldnick.com/2014/01/13/hiding-implementation-details-with-e6-weakmaps.html)，作者：尼克·菲茨杰拉德(2014)
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 2]
```diff
@@ -5,99 +5,253 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
-
+1. 首先赋值和搜索操作都是 `O(n)` 的时间复杂度（_n_ 是键值对的个数），因为这两个操作都需要遍历全部整个数组来进行匹配。
 2. 另外一个缺点是可能会导致内存泄漏，因为数组会一直引用着每个键和值。这种引用使得垃圾回收算法不能回收处理他们，即使没有其他任何引用存在了。
 
-相比之下，原生的 `WeakMap` 持有的是每个键对象的“弱引用”，这意味着在没有其他引用存在时垃圾回收能正确进行。原生 `WeakMap` 的结构是特殊且有效的，其用于映射的 key *只有*在其没有被回收时才是有效的。
+相较之下，`WeakMap` 的键对象会强引用其值，直到该键对象被垃圾回收，但从那时起，它会变为弱引用。因此，`WeakMap`：
 
-**正由于这样的弱引用，`WeakMap` 的 key 是不可枚举的**（没有方法能给出所有的 key）。如果 key 是可枚举的话，其列表将会受垃圾回收机制的影响，从而得到不确定的结果。因此，如果你想要这种类型对象的 key 值的列表，你应该使用 {{jsxref("Map")}}。
+- 不会阻止垃圾回收，直到垃圾回收器移除了键对象的引用
+- 允许垃圾回收任何值，只要它们的键对象没有其他的引用存在于 `WeakMap` 以外
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:30]:
```suggestion
- 任何值都可以被垃圾回收，只要它们的键对象没有被 `WeakMap` 以外的地方引用
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 3]
```diff
@@ -5,99 +5,253 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
 
 ## 构造函数
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:32]: 少了两段

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 4]
```diff
@@ -5,99 +5,253 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
 wm1.get(o2); // "azerty"
-wm2.get(o2); // undefined，wm2 中没有 o2 这个键
-wm2.get(o3); // undefined，值就是 undefined
+wm2.get(o2); // undefined，设置的值就是 undefined
+wm2.get(o3); // undefined，wm2 中没有 o3 这个键
+
 wm1.has(o2); // true
-wm2.has(o2); // false
-wm2.has(o3); // true (即使值是 undefined)
+wm2.has(o2); // true (即使值是 undefined)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:80]:
```suggestion
wm2.has(o2); // true（即使值是 undefined）
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 5]
```diff
@@ -5,99 +5,253 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
-    this._wm.set(k, v);
+    this.#wm.set(k, v);
     return this;
   }
 }
 ```
 
+### 模拟私有成员
+
+开发者可以使用 {{jsxref("WeakMap")}} 关联对象与私有数据，从而获得以下好处：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:120]: 移除英文版本的链接

```suggestion
开发者可以使用 `WeakMap` 关联对象与私有数据，从而获得以下好处：
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 6]
```diff
@@ -5,99 +5,253 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
+thing.showPublic();
+// "foo"
+
+thing.showPrivate();
+// 1
+```
+
+### 关联元数据
+
+{{jsxref("WeakMap")}} 可用于将元数据与对象关联，而不影响对象的生命周期。这与私有成员示例非常相似，因为私有成员也是以外部的形式模拟的元数据，不参与[原型继承](/zh-CN/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:195]:
```suggestion
`WeakMap` 可用于将元数据与对象关联，而不影响对象的生命周期。这与私有成员示例非常相似，因为私有成员也是以外部的形式模拟的元数据，不参与[原型继承](/zh-CN/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)。
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 7]
```diff
@@ -106,11 +260,11 @@ class ClearableWeakMap {
...
 
-## 相关链接
+## 参见
 
-- A polyfill of `WeakMap` is available in [`core-js`](https://github.com/zloirock/core-js#weakmap)
-- [`WeakMap` in the JavaScript guide](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#WeakMap_object)
-- [Hiding Implementation Details with ECMAScript 6 WeakMaps](http://fitzgeraldnick.com/weblog/53/)
+- [`core-js` 中 `WeakMap` 的 polyfill](https://github.com/zloirock/core-js#weakmap)
+- [带键的集合](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap_对象)
+- [使用 ECMAScript 6 WeakMap 隐藏实现细节](https://fitzgeraldnick.com/2014/01/13/hiding-implementation-details-with-e6-weakmaps.html)，作者：尼克·菲茨杰拉德(2014)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:267]:
```suggestion
- [使用 ECMAScript 6 WeakMap 隐藏实现细节](https://fitzgeraldnick.com/2014/01/13/hiding-implementation-details-with-e6-weakmaps.html)，作者：尼克·菲茨杰拉德（2014）
```

**Review by yin1999 — COMMENTED** [review]
Thank you @JasonLamv-t. I've left some suggestions here:

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 8]
```diff
@@ -5,99 +5,257 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
+1. 首先赋值和搜索操作都是 `O(n)` 的时间复杂度（_n_ 是键值对的个数），因为这两个操作都需要遍历全部整个数组来进行匹配。
 2. 另外一个缺点是可能会导致内存泄漏，因为数组会一直引用着每个键和值。这种引用使得垃圾回收算法不能回收处理他们，即使没有其他任何引用存在了。
 
-相比之下，原生的 `WeakMap` 持有的是每个键对象的“弱引用”，这意味着在没有其他引用存在时垃圾回收能正确进行。原生 `WeakMap` 的结构是特殊且有效的，其用于映射的 key *只有*在其没有被回收时才是有效的。
+相较之下，`WeakMap` 的键对象会强引用其值，直到该键对象被垃圾回收，但从那时起，它会变为弱引用。因此，`WeakMap`：
+
+- 不会阻止垃圾回收，直到垃圾回收器移除了键对象的引用
+- 任何值都可以被垃圾回收，只要它们的键对象没有被 `WeakMap` 以外的地方引用
+
+当将键映射到与键相关的信息，而该信息*仅*在键未被垃圾回收时具有价值时，`WeakMap` 可以是一个特别有用的构造。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:32]:
```suggestion
当将键映射到与键相关的信息，而该信息*仅*在键未被垃圾回收时具有价值时，`WeakMap` 是一个特别有用的构造。
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md (hunk 5 above)
Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:124]: https://github.com/mdn/content/pull/29929

Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md]: already merge into upstream

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md [hunk 9]
```diff
@@ -5,99 +5,257 @@ slug: Web/JavaScript/Reference/Global_Objects/WeakMap
...
+1. 首先赋值和搜索操作都是 `O(n)` 的时间复杂度（_n_ 是键值对的个数），因为这两个操作都需要遍历全部整个数组来进行匹配。
 2. 另外一个缺点是可能会导致内存泄漏，因为数组会一直引用着每个键和值。这种引用使得垃圾回收算法不能回收处理他们，即使没有其他任何引用存在了。
 
-相比之下，原生的 `WeakMap` 持有的是每个键对象的“弱引用”，这意味着在没有其他引用存在时垃圾回收能正确进行。原生 `WeakMap` 的结构是特殊且有效的，其用于映射的 key *只有*在其没有被回收时才是有效的。
+相较之下，`WeakMap` 的键对象会强引用其值，直到该键对象被垃圾回收，但从那时起，它会变为弱引用。因此，`WeakMap`：
+
+- 不会阻止垃圾回收，直到垃圾回收器移除了键对象的引用
+- 任何值都可以被垃圾回收，只要它们的键对象没有被 `WeakMap` 以外的地方引用
+
+当将键映射到与键相关的信息，而该信息*仅*在键未被垃圾回收时具有价值时，`WeakMap` 是一个特别有用的构造。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/index.md:32]:
```suggestion
当将键映射到与键相关的信息，而该信息*仅*在键未被垃圾回收的情况下具有价值时，`WeakMap` 是一个特别有用的构造。
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**